finetune/adapter.py

```diff
@@ -61,7 +62,9 @@ def main(
     data_dir: str = "data/alpaca", 
     pretrained_path: str = "checkpoints/lit-llama/7B/lit-llama.pth",
     out_dir: str = "out/adapter/alpaca",
+    is_instruction_tuning: bool = True,
```

Comment:
I propose to only go one way. Either exposing the variable as a global above, or making it configurable, but not both. I vote for the global variable. Any thoughts?

Reply:
I left it as a mixture of both to not have to pass the variable through multiple layers of functions in code (from `main` to `train` to `validate` to `generate_response`) if it was done as only a configurable CLI flag, maybe a future refactor could put all of these into some "Context" dataclass that contains those hyperparameters rather than passing each specific variable, especially as most of the globals are the same names across all three finetuning scripts

Reply:
We're trying to limit the amount of CLI arguments as much as we can.
I would propose to keep this as a global and not as a CLI argument for the time being.